### DISCUSSION TITLE
How to overwrite baseUrl via commandline when using a config file that extends cypress.json?

### DISCUSSION THREAD

**Discussion by timnederhoff** [cypress-io/cypress#16367] [CI setup] [answered]
In my usecase I use the cypress.json config, and extends it in separate cypress-e2e.json and cypress-smoke.json to have different baseUrls for different environments. But sometimes however I want to run tests on another environment, i.e. on another baseUrl. Preferably I want to run that test from CI with the command argument "CYPRESS_BASE_URL=https://example.com cypress run... etc", but it occurs to me that the baseUrl isn't overwritten by my command line options as long as I have the cypress.json extended in one of my other config files.

Is there a workaround to have the baseUrl overwritten while using the --config-file option in combination with the extends feature?

In my example the command that doesn't overwrite baseUrl is:
`CYPRESS_BASE_URL="https://www.myexample.com" cypress run --config-file smoke-cypress.json`

ps. I also tried `CYPRESS_BASEURL` and `CYPRESS_baseUrl` but no success.

**Comment by lmiller1990** [comment, marked as answer]
Here's a neat trick. You can dynamically modify the config from `cypress/plugins/index.js`:

```js
// cypress/plugins/index.js
module.exports = (on, config) => {
  return config
}
```

So, you could do:

```js
// cypress/plugins/index.js
module.exports = (on, config) => {
  return {
    ...config
    baseUrl: process.env.CYPRESS_BASE_URL || config.baseUrl
  }
}
```

That would override the `baseUrl` dynamically from an environment variable. Since `plugins/index.js` is just JS, you have lots of power and flexibility. You should be able to use some combination of code and config to accomplish what you want.

**Reply by timnederhoff** [reply]
Thanks! will workaround with that for now.
